### PR TITLE
Add `--frontend-server-starter-path` option to `flutter run` and `flutter test`

### PR DESCRIPTION
**Pull request by derekxu16**

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -849,6 +850,17 @@ abstract class FlutterCommand extends Command<void> {
     );
   }
 
+  void usesFrontendServerStarterPathOption({required bool verboseHelp}) {
+    argParser.addOption(
+      FlutterOptions.kFrontendServerStarterPath,
+      help: 'When this value is provided, the frontend server will be started '
+            'in JIT mode from the specified file, instead of from the AOT '
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:857]: Does this need to be a dart source entrypoint, a JIT snapshot, or either?

Comment (derekxu16) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:857]: I was only considering the case of it being a Dart source entrypoint, but you're right that the AppJIT snapshot case is also supported. I've updated the help message.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:857]: SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM (verified this worked for me locally on linux)

**Pull request merged**